src/audio/NullAudioEngine.cpp

```diff
@@ -0,0 +1,41 @@
+#include "NullAudioEngine.h"
+
+using namespace Audio;
+
+namespace Audio {
+class NullAudioSound final
+    : public Sound
+{
+public:
+    void pitch(float p) override {}
```

Comment:
As some parts of interface aren't implemented you can think about splitting interface into smaller ones. Idea from SOLID. ;)

Reply:
I'm not usre I understand what do you mean? The interface is fully implemented, it just doesn't do anything :)

Reply:
Well im not cpp guy but he means that having methods from contract which are not doing anything in class which implements this contract(like no body at all or throwing exeption) is overall a bad concept and you should rethink your interfaces. It should be separated interface then which this class just shouldn't implement.

Reply:
The current way this is implemented is to allow compiling for platforms that don't support audio, or for running the game without any audio, without having to change anything about the frontend code. The NullAudioEngine provides a way to let the game engine use an AudioEngine without actually doing anything.

This is the best I could come up with, suggestions are welcome!